### PR TITLE
Add `get2_mut` and `get2_unchecked_mut`.

### LINKED ISSUE #64: Feature Request: swap
Since it's impossible to borrow two mutable references out of one slab, it's necessary to provide something like `std::mem::swap`.

### PR DESCRIPTION
**Pull request by ksqsf**
This should fix #64.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -716,6 +758,35 @@ impl<T> Slab<T> {
         }
     }
 
+    /// Return two mutable references to the values associated with the two
+    /// given keys simultaneously without performing bounds checking.
+    ///
+    /// This function should be used with care.
```
Comment (tormol) [inline src/lib.rs]: A `# Safety` section documenting the invariants that the caller must uphold (such as key1 != key2) would be nice.

**Comment by tormol** [comment]
Hello @ksqsf

The reason I looked at this PR is because two equal indexes isn't the most obvious edge case :)

Would it make sense to return a single `Option<(&mut T, &mut T)>` instead? I assume that in most cases either index being invalid is a bug, and returning a single option makes it easier to `.unwrap()`.

File: src/lib.rs [hunk 2]
```diff
@@ -716,6 +759,40 @@ impl<T> Slab<T> {
...
+    /// ```
+    pub unsafe fn get2_unchecked_mut(&mut self, key1: usize, key2: usize) -> (&mut T, &mut T) {
+        let ptr1 = self.entries.get_mut(key1).map(|r| r as *mut Entry<T>);
+        let ptr2 = self.entries.get_mut(key2).map(|r| r as *mut Entry<T>);
+        match (ptr1.map(|p| &mut *p), ptr2.map(|p| &mut *p)) {
+            (
+                Some(&mut Entry::Occupied(ref mut val1)),
+                Some(&mut Entry::Occupied(ref mut val2)),
+            ) => (val1, val2),
+            _ => unreachable!(),
```
[the remark this answers is not in the document]
Comment (ksqsf) [reply src/lib.rs]: Other `_unchecked` functions do not say this, so I think it's better to match them.

**Comment by ksqsf** [comment]
Hi @tormol 
Thanks for your review. Just committed the changes.

File: src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tormol) [reply src/lib.rs]: Makes sense then. (I didn't check)

**Comment by carllerche** [comment]
@tormol  You are a +1 on the PR?

**Comment by tormol** [comment]
Yes

**Review by taiki-e — APPROVED** [review]

**Pull request merged**